h/views/api/annotations.py

```diff
@@ -216,6 +217,11 @@ def _record_search_api_usage_metrics(
 
     for k in keys:
         if k in params:
+            # The custom_parameter function only retains the first 255 characters
+            # of string types so convert these to utf-8 & skip chars that can't be.
+            value = params[k]
+            if k not in ["offset", "sort", "_separate_replies"]:
+                value = native(params[k], "utf-8", "ignore")
             # The New Relic Query Language does not permit _ at the begining
             # and offset is a reserved key word.
-            record_param("es_{}".format(k), str(params[k]))
```

Comment:
Was it necessary to cast the value using `str(...)` in the first place? The [New Relic documentation](https://docs.newrelic.com/docs/agents/python-agent/python-agent-api/add_custom_parameter) states that the _value_ argument to `add_custom_parameter` may be a string, integer, float or boolean.

I believe the type of `params[k]` will be a `unicode` here (in Python 2)? If so, it looks like newrelic can already handle that internally.

Reply:
Ok I gave this a try it looks like you are correct-it's not necessary so I just removed the str casting.